go/private/actions/library.bzl

```diff
@@ -49,13 +50,13 @@ def emit_library(ctx, go_toolchain,
     cgo_info_label = None
   for t in embed:
     goembed = t[GoEmbed]
-    direct_archives += get_archive(t).direct
+    direct_archives = sets.union(direct_archives, get_archive(t).direct)
```

Comment:
Same comment as earlier: use the new depset constructor instead of union in a loop.

Also in the loops below.